service/storage_proxy.cc

```diff
@@ -5018,11 +5029,14 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
         auto fence = storage_proxy::get_fence(*_effective_replication_map_ptr);
         if (_proxy->is_me(ep)) {
             tracing::trace(_trace_state, "read_mutation_data: querying locally");
-            return _proxy->apply_fence(_proxy->query_mutations_locally(_schema, cmd, _partition_range, timeout, _trace_state), fence, _proxy->my_address());
+            co_return co_await _proxy->apply_fence(_proxy->query_mutations_locally(_schema, cmd, _partition_range, timeout, _trace_state), fence, _proxy->my_address());
```

Comment:
This continuation->coroutine transform in a hot path has to be deliberate to consider performance. Please avoid it for now, we can consider it later if it doesn't regress performance.